src/key.rs

```diff
@@ -418,6 +443,20 @@ impl PublicKey {
         }
     }
 
+    /// Creates a [`PublicKey`] using the key material from `pk` combined with the `parity`.
+    pub fn from_xonly(pk: XOnlyPublicKey, parity: Parity) -> Result<PublicKey, Error> {
```

Comment:
I am sorry to bring this up again. But I think this should add `0x02` by default. `XOnlyPublicKey` is a full public key with `0x02` y co-ordinate.

Reply:
I am unable to make forward progress on this PR, @sanket1729  and @apoelstra currently have opposing views, how do we want to move forward, with or without parity parameter?

Reply:
I concede, let's go with @sanket1729's suggestion.